frontend/views/Home/Admins.tsx

```diff
@@ -21,61 +51,165 @@ type AdminProps = {
+  const { user, theme, token, admins, onRefresh } = props;
+  const [deleteModal, setDeleteModal] = useState(false);
+  const [editModal, setEditModal] = useState(false);
+  const [selectedItem, setSelectedItem] = useState(null);
 
-  const onDelete = async (email: string) => {
+  const name = useRef();
+
+  const {
+    values,
+    handleChange,
+    handleBlur,
+    errors,
+    touched,
+    handleSubmit,
+    validateField,
+    setFieldValue,
+  } = useFormik({
+    validationSchema: EmailNameValidation,
+    initialValues: initialValues,
+    onSubmit: (values) => onEdit(values),
+  });
+
+  const onDelete = async () => {
     try {
-      await Admin.delete(email, token);
+      await Admin.delete(selectedItem.email, token);
     } catch {
       // show error message
     }
     onRefresh();
   };
 
-  const onEdit = (user: {
-    email: string;
-    name: string;
-    superUser: boolean;
-  }) => {
-    //TODO: update Admin.update() params
+  const onEdit = async (fields: EditFields) => {
+    const { name, email } = fields;
     try {
-      //Admin.update(email, token)
+      Admin.update(email, name, props.token);
+      setEditModal(false);
       onRefresh();
     } catch {
-      // show error message
+      // TODO: show error message
     }
   };
 
+  const superUserDeleteCheck = (item: any) => {
+    if (!item.superUser && item.email != user.email)
+      return (
+        <IconButton
+          icon="delete"
+          onPress={() => {
+            setDeleteModal(true);
+            setSelectedItem(item);
+          }}
+        />
+      );
+  };
+
+  const handleEditClick = (admin: Admin) => {
+    setFieldValue("name", admin.name);
+    setFieldValue("email", admin.email);
+    setEditModal(true);
+  };
+
   return (
-    <View style={{ flex: 1 }}>
-      <FlatList
-        data={admins}
-        keyExtractor={(_, index) => index.toString()}
-        renderItem={({ item }) => {
-          return (
+    <FlatList
+      style={{ flex: 1 }}
+      data={admins}
+      keyExtractor={(_, index) => index.toString()}
+      renderItem={({ item }) => {
+        return (
+          <View>
             <List.Item
               title={item.email}
-              onPress={() => {
-                Alert.alert("user pressed", item.name, [
-                  { text: "OK", onPress: () => console.log("OK Pressed") },
-                ]);
-              }}
+              onPress={() => {}}
               right={() =>
                 props.token !== "" && (
                   <View style={{ flexDirection: "row" }}>
-                    <IconButton icon="pencil" onPress={() => onEdit(item)} />
                     <IconButton
-                      icon="delete"
-                      onPress={() => onDelete(item.email)}
+                      icon="pencil"
+                      onPress={() => handleEditClick(item)}
                     />
+                    {superUserDeleteCheck(item)}
```

Comment:
Should be refactored to use Component syntax, have it take the item as props and pass it as

```tsx
<SuperUserDeleteCheck item={item} />
```